modules/lib_ic/src/ic_low_level.c

```diff
@@ -282,88 +281,77 @@ float_s32_t float_s32_add_fix(
   return res;
 }
 
-//Python port
-void ic_adaption_controller(ic_state_t *state, uint8_t vad){
+//Calculates fast ratio
+void ic_calc_fast_ratio(ic_adaption_controller_state_t * ad_state){
+    const float_s32_t delta = {7037, -46}; // ~ 0.0000000001 from py_ic
+    float_s32_t denom = float_s32_add_fix(ad_state->input_energy, delta);
+    ad_state->fast_ratio = float_s32_div(ad_state->output_energy, denom);
+}
+
+void ic_set_mu(ic_state_t * state, float_s32_t mu){
+    for(int ych=0; ych<IC_Y_CHANNELS; ych++) {
+        for(int xch=0; xch<IC_X_CHANNELS; xch++) {
+            state->mu[ych][xch] = mu;
+        }
+    }
+}
+
+void ic_mu_control_system(ic_state_t * state, float_s32_t vnr){
     ic_adaption_controller_state_t *ad_state = &state->ic_adaption_controller_state;
     ic_adaption_controller_config_t *ad_config = &state->ic_adaption_controller_state.adaption_controller_config;
-
+    
     if(!ad_config->enable_adaption_controller){ //skip this function if adaption controller not enabled
         return;
     }
 
-    const float_s32_t one = {1, 0};
-    const float_s32_t zero = {0, 0};
-    //const float_s32_t delta = {1100, -40}; //1100 * 2**-40 = 0.000000001 (from stage_b.py)
-    // in some test cases delta = 1e-9 was not small enough comparing to the denominator
-    // this is also changed in stage_b test python
-    const float_s32_t delta = {352, -45}; //352 * 2**-45 ~ 0.00000000001
-
-    exponent_t q0_8_exp = -8; 
-    float_s32_t vad_float = {vad, q0_8_exp}; //convert to float between 0 and 0.99609375
-
-    //self.smoothed_voice_chance = self.voice_chance_alpha*self.smoothed_voice_chance
-    //self.smoothed_voice_chance = max(self.smoothed_voice_chance, vad_result)
-    ad_state->smoothed_voice_chance = float_s32_mul(ad_state->smoothed_voice_chance, ad_config->voice_chance_alpha);
-    if(float_s32_gt(vad_float, ad_state->smoothed_voice_chance)){
-        ad_state->smoothed_voice_chance = vad_float;
-    }
-
-    //noise_mu = 1.0 - self.smoothed_voice_chance
-    float_s32_t noise_mu = float_s32_sub(one, ad_state->smoothed_voice_chance);
+    float_s32_t one = {1, 0};
 
-    //noise_mu = noise_mu * min(1.0, np.sqrt(np.sqrt(self.output_energy/(self.input_energy + delta))))
-    float_s32_t input_plus_delta = float_s32_add_fix(ad_state->input_energy_slow, delta);
-    float_s32_t ratio = float_s32_div(ad_state->output_energy_slow, input_plus_delta);
-    ratio = float_s32_sqrt(ratio);
-    ratio = float_s32_sqrt(ratio);
-    if(float_s32_gt(one, ratio)){ 
-        ratio = one;
+    if(float_s32_gte(vnr, ad_config->input_vnr_threshold)){
```

Comment:
Is there no code corresponding to the python below in C?

elif self.adaption_config == "IC_ADAPTION_FORCE_OFF":
            self.set_mu(0.0)
            self.set_leakage(1.0)
            self.control_flag = control_state.FORCE_HOLD
            return self.mu, self.control_flag